File: LeetCode/[M] - 334. Increasing Triplet Subsequence/solution.py
class Solution:
    def increasingTriplet(self, nums: list[int]) -> bool:
        smlst1, smlst2 = float("inf"), float("inf")
        for num in nums:
            if num > smlst2:
                return True
            if num > smlst1:
                smlst2 = num
            else:
                smlst1 = num
        return False

File: LeetCode/[M] - 334. Increasing Triplet Subsequence/solution.go
package main

import "math"

func increasingTriplet(nums []int) bool {
	smlst1, smlst2 := int(math.Pow(2, 31)), int(math.Pow(2, 31))
	for _, el := range nums {
		switch {
		case el > smlst2:
			return true
		case el > smlst1:
			smlst2 = el
		default:
			smlst1 = el
		}
	}
	return false
}
